algopy_testing/src/algopy_testing/arc4.py

```diff
@@ -59,6 +58,55 @@ def from_log(cls, log: algopy.Bytes, /) -> typing.Self:
         ...
 
 
+class String(_ABIEncoded):
+    """An ARC4 sequence of bytes containing a UTF8 string"""
+
+    _value: bytes
+
+    def __init__(self, value: algopy.String | str = "", /) -> None:
+        value = as_string(value)
+        bytes_value = value.encode("utf-8")
```

Comment:
To preserve semantic compatibility between these stubs and the compiled behaviour this should only encode value to bytes for `str` type, for the `algopy.String` type it should use the underlying bytes of that type, this probably means changing the `algopy.String` implementation too to support this. Any other type should be a TypeError

Reply:
It has been fixed in this commit daab1b78c560ff747c4569d4f44321ad180e70f7.